Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -233,31 +233,65 @@ extension _ChatChannelListVC: _ChatChannelListControllerDelegate {
         _ controller: _ChatChannelListController<ExtraData>,
         didChangeChannels changes: [ListChange<_ChatChannel<ExtraData>>]
     ) {
-        let newChannelsCount = controller.channels.count
-        var movedItems: [IndexPath] = []
-        
+        var movedItems = Set<Range<IndexPath>>()
+        var insertedItems = Set<IndexPath>()
+        var removedItems = Set<IndexPath>()
+        var updatedItems = Set<IndexPath>()
+
+        for change in changes {
+            switch change {
+            case let .insert(_, index):
+                insertedItems.insert(index)
+            case let .move(_, fromIndex, toIndex):
+                movedItems.insert(Range(uncheckedBounds: (fromIndex, toIndex)))
+            case let .remove(_, index):
+                removedItems.insert(index)
+            case let .update(_, index):
+                updatedItems.insert(index)
+            }
+        }
+
+        let moveFromItems = Set(movedItems.map(\.lowerBound))
+        let moveToItems = Set(movedItems.map(\.upperBound))
+
+        let numberOfUpdates = moveFromItems.count
+            + moveToItems.count
+            + insertedItems.count
+            + removedItems.count
+            + updatedItems.count
+
+        let mergedItems = moveFromItems
+            .union(moveToItems)
+            .union(insertedItems)
+            .union(removedItems)
+            .union(updatedItems)
+
+        // If doing an union of all IndexPath changes and the count is less than before
+        // It means there are duplicates, and so we have conflicting updates
+        let hasConflictingUpdates = mergedItems.count != numberOfUpdates
+
+        if hasConflictingUpdates {
+            collectionView.reloadData()
+
+            channelsCount = controller.channels.count
+            return
+        }
```

Comment:
Can you move all this to a separate method/extension? Also, if the updates are conflicting updates, can we do something like:
```
log.error("⚠️ Inconsistent updates from ChatChannelListController. Please copy the following log and open an issue at https://github.com/GetStream/stream-chat-swift/issues")
+ pretty print the changes
```